Libraries/NativeAnimation/RCTAnimatedValueChangeEvent.m

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTAnimatedValueChangeEvent.h"
+
+@implementation RCTAnimatedValueChangeEvent
+{
+  CGFloat _value;
+}
+
+@synthesize viewTag = _viewTag;
+@synthesize eventName = _eventName;
+@synthesize coalescingKey = _coalescingKey;
+
+- (instancetype)initWithNodeTag:(NSNumber *)nodeTag
+                          value:(CGFloat)value
+{
+  if ((self = [super init])) {
+    _eventName = @"onAnimatedValueUpdate";
+    _viewTag = nodeTag;
```

Comment:
RCTEvent protocol is a bit specific to views events but the coalescing logic can be used for any kind of event. Ideally this would not be called viewTag but for now we use this to coalesce events by node tag. It would have also been possible to use coalescingKey instead.

Reply:
How can we make it less specific or simply better? Let's do this. 😄